scripts/generate-deploy-yamls.sh

```diff
@@ -1,52 +1,158 @@
+			exit 0
+      ;;
+    -o)
+      shift
+      output_dir=$1
+      ;;
+    -p)
+      shift
+      pools="$pools $1"
+      ;;
+    -r)
+      shift
+      registry=$1
+      ;;
+    -t)
+      shift
+      tag=$1
+      ;;
+    -*)
+      echo "Unknown option: $1"
+      help
+      exit 1
+      ;;
+    *)
+      profile=$1
+			shift
+			break
+      ;;
+  esac
+  shift
+done
 
-if [ "$1" = "-t" ]; then
-	TARGET_DIR="$2"
-	shift 2
-else
-	TARGET_DIR="$SCRIPTDIR/../deploy"
-fi
-if [ ! -d "$TARGET_DIR" ]; then
-	mkdir -p "$TARGET_DIR"
+# The space after profile name is reserved for profile specific options which
+# we don't have yet.
+if [ "$#" -gt 0 ]; then
+	help
+	exit 1
 fi
 
-if [ "x$2" = x ]; then
-	mayastor_images_repo="NONE"
+# In most of the cases the tag will be a specific version that does not change
+# so save dockerhub bandwidth and don't always pull the image.
+if [ -n "$tag" ]; then
+	pull_policy=IfNotPresent
 else
-	mayastor_images_repo="$2"
+	pull_policy=Always
 fi
 
+# Set profile defaults
+case "$profile" in
+  "develop")
+		[ -z "$cores" ] && cores=1
+		[ -z "$tag" ] && tag=develop
+		moac_debug="-vv"
+	  ;;
+  "release")
+		[ -z "$cores" ] && cores=1
+		[ -z "$tag" ] && tag=latest
+	  ;;
+  "test")
+		[ -z "$cores" ] && cores=2
+		[ -z "$tag" ] && tag=ci
+	  ;;
+  *)
+		echo "Missing or invalid profile name. Type \"$0 --help\""
+		exit 1
+	  ;;
+esac
+
 set -u
 
-if ! which helm > /dev/null 2>&1; then
-	echo "Install helm to path >v3.4.1"
-	echo "https://github.com/helm/helm/releases/tag/v3.4.1"
+if ! which helm >/dev/null 2>&1; then
+	echo "Install helm (>v3.4.1) to PATH"
 	exit 1
 fi
 
+if [ ! -d "$output_dir" ]; then
+	mkdir -p "$output_dir"
+fi
+
 tmpd=$(mktemp -d /tmp/generate-deploy.sh.XXXXXXXX)
 # shellcheck disable=SC2064
 trap "rm -fr '$tmpd'" HUP QUIT EXIT TERM INT
 
-template_params="mayastorImagesTag=$1"
-if [ "$mayastor_images_repo" != "NONE" ]; then
-	template_params="$template_params,mayastorImagesRepo=$mayastor_images_repo"
+# A rule of thumb: # of cpu cores equals to Gigs of hugepage memory
+template_params="mayastorImagesTag=$tag"
+template_params="$template_params,mayastorImagePullPolicy=$pull_policy"
+template_params="$template_params,mayastorCpuCount=$cores"
+template_params="$template_params,mayastorHugePagesGiB=$cores"
+template_params="$template_params,moacDebug=$moac_debug"
+if [ -n "$registry" ]; then
+	template_params="$template_params,mayastorImagesRepo=$registry"
 fi
+if [ -n "$pools" ]; then
+  i=0
+	for pool in $pools; do
+	  node=$(echo "$pool" | sed 's/,.*//')
+	  device=$(echo "$pool" | sed 's/.*,//')
+		template_params="$template_params,mayastorPools[$i].node=$node"
+		template_params="$template_params,mayastorPools[$i].device=$device"
+		i=$((i + 1))
+	done
+fi
+
 helm template --set "$template_params" mayastor "$SCRIPTDIR/../chart" --output-dir="$tmpd" --namespace mayastor
 
-mv "$tmpd"/mayastor/templates/*.yaml "$TARGET_DIR"
+cd "$tmpd/mayastor/templates"
+mv ./*.yaml "$output_dir/"
```

Comment:
Please don't change working directory in scripts. Working directory should be there for the user, not scripts. Moreover it's like using global variables/mutating global state. I don't even understand the need here. Sooner or later it always causes problems to someone (e.g. continuing this script all relative paths are now broken). If you really need to do this (which I don't understand and don't see a reason) please use at least a subshell.
`( cd /whatever && do_stuff )`

Reply:
The reason for this was basically because I wanted to avoid string quotation like this:
```
mv "$tmpd"/mayastor/templates/*.yaml ...
```
Note that I cannot quote the whole string as usually because that will not expand to multiple files:
```
mv "$tmpd/mayastor/templates/*.yaml" ...   <--- does not work
```
So either I have to quote just the variable and not the whole string, or I have to change the directory first and then do the copy (and I can do that in subshell - no problem). I'm fine with either option. Let me know what you would prefer.

Reply:
I prefer leaving it as it is as is perfectly correct shell code :smile:. Second best is the subshell.

When at it (just brainstorm, you made me think :smile:) `find "$tmpd/mayastor/templates" -name '*.yaml' | xargs mv -t "$output_dir"` is also valid and doesn't contain partial path quoting.

FYI just to get the feel for it I did a small research over my `/usr/bin` and partial quoting is ubiquitous "technique":
```
file /usr/bin/* | grep shell | cut -f1 -d: | xargs grep --color=always '"/' 
```